src/CommunityToolkit.Aspire.Hosting.Umami/UmamiBuilderExtensions.cs

```diff
@@ -0,0 +1,93 @@
+    /// </summary>
+    /// <param name="builder">The <see cref="IDistributedApplicationBuilder"/>.</param>
+    /// <param name="name">The name of the resource. This name will be used as the connection string name when referenced in a dependency.</param>
+    /// <param name="secret">The parameter used to provide the app secret for the Umami resource.</param>
+    /// <param name="port">The host port for the Umami app.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
+    /// <remarks>
+    /// <example>
+    /// Add a Umami container to the application model and reference it in a .NET project.
+    /// <code lang="csharp">
+    /// var builder = DistributedApplication.CreateBuilder(args);
+    ///
+    /// var db = builder.AddPostgres("postgres")
+    ///   .AddDatabase("db");
+    /// var umami = builder.AddUmami("umami")
+    ///   .WithStorageBackend(db);
+    ///  
+    /// builder.Build().Run(); 
+    /// </code>
+    /// </example>
+    /// </remarks>
+    public static IResourceBuilder<UmamiResource> AddUmami(
+        this IDistributedApplicationBuilder builder,
+        [ResourceName] string name,
+        IResourceBuilder<ParameterResource>? secret = null,
+        int? port = null
+    )
+    {
+        ArgumentNullException.ThrowIfNull(builder);
+        ArgumentException.ThrowIfNullOrEmpty(name);
+
+        // The secret must be at least 8 characters long and contain characters from three of the following four sets: Uppercase letters, Lowercase letters, Base 10 digits, and Symbols
+        var secretParameter = secret?.Resource ?? ParameterResourceBuilderExtensions.CreateDefaultPasswordParameter(builder, $"{name}-secret", minLower: 1, minUpper: 1, minNumeric: 1);
+
+        var umami = new UmamiResource(name, secretParameter);
+
+        return builder.AddResource(umami)
+            .WithHttpEndpoint(port: port, targetPort: UmamiPort, name: UmamiResource.PrimaryEndpointName)
+            .WithImage(UmamiContainerImageTags.Image, UmamiContainerImageTags.Tag)
+            .WithImageRegistry(UmamiContainerImageTags.Registry)
+            .WithEnvironment(context =>
+            {
+                context.EnvironmentVariables[SecretEnvVarName] = umami.SecretParameter;
+            })
+            .WithHttpHealthCheck("/api/heartbeat");
+    }
+
+    /// <summary>
+    /// References a <see cref="PostgresDatabaseResource"/> as the storage backend for the <see cref="UmamiResource"/>.
+    /// </summary>
+    /// <param name="builder">The Umami resource builder.</param>
+    /// <param name="database">The PostgreSQL database resource builder.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<UmamiResource> WithStorageBackend(
+        this IResourceBuilder<UmamiResource> builder, 
+        IResourceBuilder<PostgresDatabaseResource> database
```

Comment:
It looks like they've just dropped support for MySQL and now only support PostgreSQL, so maybe the method name could be more like `WithPostgreSQL` or something rather than trying to be generic.

Reply:
When I read their docs (admittedly a brief read) it seemed like they only support PG as a backend, v3 dropped support for MySQL

Reply:
If they bring it back, we'll handle the API breaking change then 😝